cpp/include/cudf/utilities/type_dispatcher.hpp

```diff
@@ -419,5 +419,53 @@ CUDA_HOST_DEVICE_CALLABLE constexpr decltype(auto) type_dispatcher(cudf::data_ty
   }
 }
 
+namespace {
+template <typename T1>
+struct dispatch_second_type_impl {
+#pragma nv_exec_check_disable
+  template <typename T2, typename F, typename... Ts>
+  CUDA_HOST_DEVICE_CALLABLE decltype(auto) operator()(F&& f, Ts&&... args) const noexcept
+  {
+    return f.template operator()<T1, T2>(std::forward<Ts>(args)...);
+  }
+};
+
+template <template <cudf::type_id> typename IdTypeMap>
+struct dispatch_second_type {
```

Comment:
I think naming would be more intuitive to call this one `dispatch_first_type`, and the other one `dispatch_second_type`.